torchx/specs/api.py

```diff
@@ -923,6 +934,34 @@ def __repr__(self) -> str:
 
         return out
 
+    @classmethod
+    # pyre-fixme[11]: TypedDict is not defined as a type
+    def from_typed_dict(cls, typed_dict: Type[TypedDict]) -> "runopts":
+        opts = cls()
+        doc = docstring_parser.parse(typed_dict.__doc__)
+        field_docs = {field.arg_name: field.description for field in doc.params}
+        for name, field_type in typed_dict.__annotations__.items():
+            default = getattr(typed_dict, name, None)
+            if isinstance(default, Field):
+                default_factory = default.default_factory
+                if default_factory is dataclasses.MISSING:
+                    default = None
+                else:
+                    # pyre-fixme[29]: not a function
+                    default = default_factory()
+            required = not (
```

Comment:
this is going to get interesting in python 3.11 with [PEP-0655](https://peps.python.org/pep-0655/) that introduces [`typing.Required`](https://docs.python.org/3/library/typing.html#typing.Required) and [`typing.NotRequired`](https://docs.python.org/3/library/typing.html#typing.NotRequired). 

A couple of interesting questions:

1. how do we interpret a `Required[Optional[int]]`?  (disallow it? allow "required" Optional cfgs? or implicitly make this required?)
2. are there differences between declaring a cfg as `Optional[int]` versus `NotRequired[int]`? (technically the former would be in the runopt as `None` by default, and the later would simply not exist in runopt <- this concept doesn't exist in runopt yet so we'd have to default it to `None`, but then the typing breaks)